spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/listener/TaskBatchExecutionListenerTests.java

```diff
@@ -144,38 +144,38 @@ public void testAutobuiltDataSourceNoJob() {
 
 	@Test
 	public void testMapBased() {
-		this.applicationContext = SpringApplication.run(new Object[] {JobConfiguration.class,
-				PropertyPlaceholderAutoConfiguration.class,
+		this.applicationContext = SpringApplication.run(new Class[] {JobConfiguration.class,
+				PropertyPlaceholderAutoConfiguration.class, EmbeddedDataSourceConfiguration.class,
 				BatchAutoConfiguration.class,
 				TaskBatchAutoConfiguration.class}, ARGS);
 
 		TaskExplorer taskExplorer = this.applicationContext.getBean(TaskExplorer.class);
 
-		Page<TaskExecution> page = taskExplorer.findTaskExecutionsByName("application", new PageRequest(0, 1));
+		Page<TaskExecution> page = taskExplorer.findTaskExecutionsByName("application", PageRequest.of(0, 1));
 
 		Set<Long> jobExecutionIds = taskExplorer.getJobExecutionIdsByTaskExecutionId(page.iterator().next().getExecutionId());
 
 		assertEquals(1, jobExecutionIds.size());
-		assertEquals(0, (long) taskExplorer.getTaskExecutionIdByJobExecutionId(jobExecutionIds.iterator().next()));
+		assertEquals(1, (long) taskExplorer.getTaskExecutionIdByJobExecutionId(jobExecutionIds.iterator().next()));
```

Comment:
By adding the EmbeddedDataSourceConfiguration to the context it sets the initial id start to 1.